gnd/src/main/java/com/google/android/gnd/workers/TileDownloadWorker.java

```diff
@@ -150,27 +153,23 @@ private Result checkDownload(Tile tile) {
   @NonNull
   @Override
   public Result doWork() {
-    Tile tile = localDataStore.getTile(tileId).blockingGet();
+    ImmutableList<Tile> pendingTiles = localDataStore.getPendingTiles().blockingGet();
 
-    // When there is no tile in the db, the Maybe completes and returns null.
+    // When there are no tiles in the db, the blockingGet returns null.
     // We expect tiles to be added to the DB prior to downloading.
     // If that isn't the case, we fail.
-    if (tile == null) {
+    if (pendingTiles == null) {
       return Result.failure();
     }
 
-    Log.d(TAG, "Downloading tile: " + tile.getPath());
-
-    switch (tile.getState()) {
-      case DOWNLOADED:
-        return checkDownload(tile);
-      case PENDING:
-      case FAILED:
-        return downloadTile(tile);
-      case IN_PROGRESS:
-        return resumeTileDownload(tile);
-      default:
-        return Result.failure();
+    Log.d(TAG, "Downloading tiles: " + pendingTiles);
```

Comment:
Fyi: After Shobhit's recent PR, you can now just `Timber.d("Downloading tiles %s", pendingTiles)`.

Reply:
Done! See 2b81c0b

Reply:
Now it's done :) c68a03b